core/common/src/main/java/alluxio/AlluxioURI.java

```diff
@@ -448,4 +449,18 @@ public String toString() {
     }
     return sb.toString();
   }
+
+  /**
+   * Gets the path component of the {@link AlluxioURI}.
+   *
+   * @return the absolute path
+   */
+  public String getAbsolutePath() {
+    if (mUri.getAuthority() != null) {
+      return PathUtils.concatPath(System.getProperty("user.dir"),
+              mUri.getAuthority().equals(".") ? mUri.getPath() : mUri.getAuthority());
```

Comment:
`.` is technically a part of the URI path, not the authority, so this is relying on a bug in `AlluxioURI` where it gets confused by `.` and puts it in the authority. We should either fix `AlluxioURI` to completely support relative paths (preferably in a separate PR), or not support relative paths at all.